client/my-sites/theme/main.jsx

```diff
@@ -609,8 +622,12 @@ export default connect(
 		const backPath = getBackPath( state );
 		const currentUserId = getCurrentUserId( state );
 		const isCurrentUserPaid = isUserPaid( state, currentUserId );
-		const theme = getTheme( state, siteIdOrWpcom, id );
+		// Fallback to 'wpcom' source for wpcom themes on Jetpack target sites
+		const theme = getTheme( state, siteIdOrWpcom, id ) || getTheme( state, 'wpcom', id );
 		const error = theme ? false : getThemeRequestErrors( state, id, siteIdOrWpcom );
+		const isWpcomTheme = theme && theme.screenshots;
```

Comment:
Maybe we can re-arrange the logic here a bit. When do we have a `screenshots` field? If the theme was obtained from the `wpcom` Redux subtree, not from the Jetpack-specific `siteId` one. So maybe instead of indiscriminately falling back to `wpcom` when using `getTheme` two lines above, we really need a `isWpcomTheme` selector that works the same way as our [`isWporgTheme`](https://github.com/Automattic/wp-calypso/blob/master/client/state/themes/selectors.js#L289-L298) one.